plots/generic.py

```diff
@@ -0,0 +1,539 @@
+    are offered such as sorting, grouping and colorcoding on the basis of the
+    arrangement in list of spiketrains and spiketrain annotations.
+
+    :param spiketrain_list: list
+        List can either contain neo spiketrains or lists of neo spiketrains.
+    :param key_list: str | list of str
+        Annotation key(s) for which the spiketrains should be ordered.
+        When list of keys is given the spiketrains are ordered successively for
+        the keys.
+        By default the ordering by the given lists of spiketrains have priority.
+        This can be bypassed by using an empty string '' as list-key at any
+        position in the key_list.
+    :param groupingdepth: 0 | 1 | 2
+        * 0: No grouping (default)
+        * 1: grouping by first key in key_list.
+             Note that when list of lists of spiketrains are given the first
+             key is by the list identification key ''. If this is unwanted
+             the empty string '' can be placed at a different position in
+             key_list.
+        * 2: additional grouping by second key respectively
+        The groups are separated by whitespace specified in the spacing
+        parameter and optionally by a line specified by the the separatorargs.
+    :param spacing: int | [int] | [int, int]
+        Size of whitespace separating the groups in units of spiketrains.
+        When groupingdepth == 2 a list of two values can specify the distance
+        between the groups in level 1 and level 2. When only one value is given
+        level 2 spacing is set to half the spacing of level 1.
+        Default: [5, 3]
+    :param colorkey: str | int  (default 0)
+        Contrasts values of a key by color. The key can be defined by its
+        namestring or its position in key_list. Note that position 0 points to
+        the list identification key ('') when list of lists of spiketrains are
+        given, if not otherwise specified in key_list!
+    :param pophist_mode: 'color' (default) | 'total'
+         * total: One population histogram for all drawn spiketrains
+         * color: Additionally to the total population histogram,
+                  a histogram for each colored subset is drawn (see colorkey).
+    :param pophistbins: int (default 100)
+        Number of bins used for the population histogram.
+    :param right_histogram: function
+        The function gets ONE neo.SpikeTrain as argument and has to return a
+        scalar. For example the functions in the elephant.statistics module can
+        be used. (default: mean_firing_rate)
+        When a function is applied is is recommended to set the axis label
+        accordingly by using the axis handle returned by the function:
+        axhisty.set_xlabel('Label Name')
+    :param filter_function: function
+        The function gets ONE neo.SpikeTrain as argument and if the return is
+        Truethy the spiketrain is included and if the return is Falsely it is
```

Comment:
Here and in the remainder of the doc string: please phrase boolean statements it something like "...if the return is True then the spike train is included; if False, it is..."